sklearn/cluster/_spectral.py

```diff
@@ -184,6 +184,11 @@ def spectral_clustering(affinity, *, n_clusters=8, n_components=None,
           - heat kernel of the pairwise distance matrix of the samples,
           - symmetric k-nearest neighbours connectivity matrix of the samples.
 
+        .. note::
+            ``numpy.matrix`` is not supported as not recommended by numpy. See
+            https://numpy.org/doc/stable/reference/generated/numpy.matrix.html
+            for details.
```

Comment:
I believe this is just a matter of taste, but you might want to use a [Sphinx external link](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks) in this case.
Something like 
```
See `numpy matrix documentation <https://numpy.org/doc/stable/reference/generated/numpy.matrix.html>`_
```